core/src/main/java/org/radargun/stages/cache/background/StressorRecord.java

```diff
@@ -23,9 +23,10 @@
    protected final int threadId;
    protected long currentKeyId;
    protected volatile long currentOp = -1;
-   protected final List<StressorConfirmation> confirmations = new LinkedList<>();
+   protected final List<StressorConfirmation> confirmations = Collections.synchronizedList(new LinkedList<StressorConfirmation>());
```

Comment:
It seems that most of the accesses is synchronized anyway, but getLastConfirmedOperationId() which is not and synchronizedList does not help, since you need to retrieve last element anyway (btw., confirmations.getLast() should be used there).